lib/exchanges.js

```diff
@@ -0,0 +1,94 @@
+var base      = require('taskcluster-base');
+var assert    = require('assert');
+var _         = require('lodash');
+
+// Common schema prefix
+var SCHEMA_PREFIX_CONST = 'http://schemas.taskcluster.net/github/v1/';
+
+/** Build common routing key construct for `exchanges.declare` */
+var commonRoutingKey = function(options) {
+    options = options || {};
+     let routingKey = [
+        {
+          name:             'routingKeyKind',
+          summary:          "Identifier for the routing-key kind. This is " +
+                            "always `'primary'` for the formalized routing key.",
+          constant:         'primary',
+          required:         true
+        }, {
+          name:             'organization',
+          summary:          "The GitHub `organization` which had an event.",
+          maxSize:          100,
+          required:         true
+        }, {
+          name:             'repository',
+          summary:          "The GitHub `repository` which had an event.",
```

Comment:
Ooh, the hashing is a good pattern. I'm opting for s/\./%/ instead of ~ since I think that will look more familiar and be easier to recognize/remember (we've all seen uri encoding do that with spaces).